src/components/GaugeWeight.tsx

```diff
@@ -0,0 +1,103 @@
+import { BasicPool, BasicPoolsContext } from "../providers/BasicPoolsProvider"
+import React, { useContext, useRef } from "react"
+import { BigNumber } from "@ethersproject/bignumber"
+import { CircularProgress } from "@mui/material"
+import Highcharts from "highcharts"
+import HighchartsExporting from "highcharts/modules/exporting"
+import HighchartsReact from "highcharts-react-official"
+import PieChart from "highcharts-react-official"
+import { PoolAddressToGauge } from "../providers/GaugeProvider"
+
+interface Props {
+  gauges: PoolAddressToGauge
+}
+
+export type GaugeWeightData = {
+  poolAddress: string
+  poolName: string
+  gaugeRelativeWeight: BigNumber
+}
+
+export default function GaugeWeight({
+  gauges,
+  ...props
+}: Props & HighchartsReact.Props): JSX.Element {
+  HighchartsExporting(Highcharts)
+  const chartComponentRef = useRef<HighchartsReact.RefObject>(null)
+  const pools = useContext(BasicPoolsContext)
+  if (pools == undefined) return <CircularProgress color="secondary" />
+
+  const gaugesInfo = (Object.values(pools) as BasicPool[])
+    .map((pool) => {
+      const gauge = gauges[pool.poolAddress]
+      const gaugePoolAddress = pool.poolAddress
+      return gauge
+        ? {
+            poolAddress: gaugePoolAddress,
+            poolName: pool.poolName,
+            // @ts-ignore: Object is possibly 'null'.
+            gaugeRelativeWeight: gauges[gaugePoolAddress].gaugeRelativeWeight,
```

Comment:
`gauges[gaugePoolAddress]` is already defined as `const gauge` and has already been null-checked, pls update and remove ts-ignore